AUTHENTICATION.md

```diff
@@ -110,52 +110,10 @@ eksctl utils associate-iam-oidc-provider --cluster <cluster-name> --region <regi
 
 6. Create IAM Role that provider-aws will use
 
-Set environment variables that will be used in subsequent commands:
+Create IAM role with trust relationship:
 
 ```
-OIDC_PROVIDER=$(aws eks describe-cluster --name <cluster-name> --region <region> --query "cluster.identity.oidc.issuer" --output text | sed -e "s/^https:\/\///")
-```
-
-Create trust relationship for IAM role:
-
-```
-read -r -d '' TRUST_RELATIONSHIP <<EOF
-{
-  "Version": "2012-10-17",
-  "Statement": [
-    {
-      "Effect": "Allow",
-      "Principal": {
-        "Federated": "arn:aws:iam::${AWS_ACCOUNT_ID}:oidc-provider/${OIDC_PROVIDER}"
-      },
-      "Action": "sts:AssumeRoleWithWebIdentity",
-      "Condition": {
-        "StringLike": {
-          "${OIDC_PROVIDER}:sub": "system:serviceaccount:${SERVICE_ACCOUNT_NAMESPACE}:provider-aws-*"
-        }
-      }
-    }
-  ]
-}
-EOF
-echo "${TRUST_RELATIONSHIP}" > trust.json
-```
-
-> The default service account name is the provider-aws revision and changes with every provider release. The conditional above wildcard matches the default service account name in order to keep the role consistent across provider releases.
-The above policy assumes a service account name of `provider-aws-*` 
-
-Create IAM role:
-
-```
-aws iam create-role --role-name $IAM_ROLE_NAME --assume-role-policy-document file://trust.json --description "IAM role for provider-aws"
-```
-
-Associate a policy with the IAM role. This example uses `AdministratorAccess`,
-but you should select a policy with the minimum permissions required to
-provision your resources.
-
-```
-aws iam attach-role-policy --role-name $IAM_ROLE_NAME --policy-arn=arn:aws:iam::aws:policy/AdministratorAccess
+eksctl create iamserviceaccount --cluster <cluster-name> --region <region> --name="$SERVICE_ACCOUNT_NAME" --namespace="$SERVICE_ACCOUNT_NAMESPACE" --role-name="$IAM_ROLE_NAME" --role-only --attach-policy-arn="arn:aws:iam::aws:policy/AdministratorAccess" --approve
```

Comment:
Can you use a wildcard for the service account name?

The policy in the current document uses `StringLike`:

```json
        "StringLike": {
          "${OIDC_PROVIDER}:sub": "system:serviceaccount:${SERVICE_ACCOUNT_NAMESPACE}:provider-aws-*"
        }
```

Reply:
Using wildcard is not possible in `eksctl`.

Reply:
@ruzickap OK, until we can pin the service account name (https://github.com/crossplane/crossplane/issues/2295) - I think we should have a warning here about that.